parallel.go

```diff
@@ -148,6 +148,49 @@ func (r Parallel) put(do func(routing.IpfsRouting) error) error {
 	}
 }
 
+func (r Parallel) search(ctx context.Context, do func(routing.IpfsRouting) (<-chan []byte, error)) (<-chan []byte, error) {
```

Comment:
This works but I believe the better solution would be to generalize the logic in `FindProvidersAsync` to merge *any* set of channels (like we do with `get`/`GetValue`). I'm not sure how well that'll work though.

(my main concern is spinning up a bunch of goroutines, but that may be a premature optimization).

Reply:
I don't think it's worth making the code even messier than it currently is. Given that `SearchValues` in most cases will see significantly less load than `FindProviders` (which will usually get called a lot as a result of calls to SearchValue), this should be fine.

And I still need to plug validator stuff there.

Reply:
SGTM. We can always revisit if it becomes an issue.